Automattic/jetpack
# Update conditionally render connection footer with unlock

## Proposed changes:

1. Convert Connection Status Card to TypeScript
2. Conditionally render errors, warnings, or "unlock" styles for connection based on previously activated modules and modules needing user connection (more details here in the designs: p1HpG7-rPo-p2)

### Other information:


## Jetpack product discussion

P2: pbNhbs-akp-p2

## Does this pull request change what data or activity we track or use?

Yes, this PR adds some tracks properties to the connection CTAs that track any plugins broken by site or user connection when the button is clicked

## Testing instructions:

1. Checkout this branch via the Jetpack Beta plugin or your local dev environment
NOTE: It's easier to do this via jurassic ninja since your local environment has likely been connected before, and thus would make it harder to test from scratch. However you can always go [here](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/my-jetpack/src/class-initializer.php#L766) and manually return an array like this
```php
return array(
	'needs_site_connection' => array(),
	'needs_user_connection' => array(),
);
```
to simulate a new site, or you can add some plugins in those arrays to simulate any broken plugin you want
2. On a fresh site that has never been connected, go to My Jetpack and you should see this
![image](https://github.com/Automattic/jetpack/assets/65001528/eba7b00e-bdd7-4869-8092-a1c6f5d2294f)
3. Now try connecting your site and going back to My Jetpack. If you have features that require a user connection, you should see this
![image](https://github.com/Automattic/jetpack/assets/65001528/87fc7baa-2f5d-49bb-b6b1-d961443b8723)
4. If you do not, you should see this
![image](https://github.com/Automattic/jetpack/assets/65001528/36ba3d35-8c10-4f88-880e-42a919afdbf3)
5. Now disconnect your site from the `Manage` CTA next to the site connection (to make sure it works). Once the page reloads, you should see an error-based connection message now
![image](https://github.com/Automattic/jetpack/assets/65001528/4dcbce23-4c8b-4664-8115-0525c64ada1f)
6. Now connect your site AND user. The section should now look like this
![image](https://github.com/Automattic/jetpack/assets/65001528/0126a5e0-b330-4a69-b8f8-ea334fa11689)
7. Disconnect your user account, you should now see an error based user connection message
![image](https://github.com/Automattic/jetpack/assets/65001528/43f6134a-6a60-4607-b9df-1b75bbed7467)


Bonus: If you'd like, you can test out the "Connected Owner" by user connecting your site and adding a test user, and making sure that section shows up below the other two. However, that didn't really change at all so I don't think it is necessary
